### PR TITLE
Avatar-iOS: Add docs page

### PR DESCRIPTION
**Pull request by rlingineni**
#### What changed?
Adding iOS Avatar page in Markdown

### CODE REVIEW

**Comment by AlbertCarreras** [comment]
There are changes also in the Toast page. Different PR?

File: docs/docs-components/DocsSideNavigation.js [hunk 1]
```diff
@@ -35,9 +35,14 @@ export default function DocsSideNavigation(): Node {
   // If it's the components section, find the section that is currently
   // filtered with the component platform switcher
 
+  // if it's a markdown path, then the url is provieded in the query obj
```
Comment (AlbertCarreras) [inline docs/docs-components/DocsSideNavigation.js:38]: typo provided

Comment (rlingineni) [reply docs/docs-components/DocsSideNavigation.js:38]: ooh will fix!

**Comment by rlingineni** [comment]
> There are changes also in the Toast page. Different PR?

Yeah, we were just jamming. It'll get removed in a bit.

Could you merge this one before, since this PR depends on changes/resolutions from it?
https://github.com/pinterest/gestalt/pull/2227

File: docs/pages/[...id].js [hunk 2]
```diff
@@ -54,17 +58,18 @@ export async function getStaticProps(context: {| params: {| id: string |} |}): P
...
       content: mdxSource,
-      pageSourceUrl: `https://github.com/pinterest/gestalt/tree/master/docs/pages/markdown/${id}.md`,
+      pageSourceUrl: `https://github.com/pinterest/gestalt/tree/master/docs/pages/markdown/${pathName}.md`,
     },
   };
 }
 
 export async function getStaticPaths(): Promise<{|
-  paths: {| params: {| id: string |} |}[],
+  paths: {| params: {| id: string | string[] |} |}[],
```
Comment (dangerismycat) [inline docs/pages/[...id].js:67]: Before you land this, pull in the changes from #2302 (once it's merged) and re-run `yarn` to lint for the array Flow type syntax here

File: docs/utils/mdHelper.js [hunk 3]
```diff
@@ -28,13 +29,56 @@ export async function getDocByRoute(route: string): Promise<{|
...
 
-  const folderContents = await fs.readdir(docsDirectory);
-  const cleanedNames = folderContents
-    .filter((name) => name.endsWith('.mdx') || name.endsWith('.md'))
-    .map((name) => name.replace('.md', ''));
+  const getAllSitePaths = () => {
+    const pagePaths = [];
+
+    const addUrlPaths = (pageItems: Array<siteIndexType | string>, _pages: Array<string>) => {
+      const pages: Array<string> = _pages.concat([]);
```
Comment (dangerismycat) [inline docs/utils/mdHelper.js:43]: What is `.concat([])` doing here? That should be effectively a no-op?

File: docs/utils/mdHelper.js [hunk 4]
```diff
@@ -28,13 +29,56 @@ export async function getDocByRoute(route: string): Promise<{|
...
+
+      // for each choice
+      pageItems.forEach((page) => {
+        if (page.sectionName) {
+          // $FlowFixMe[incompatible-type] This is a siteIndexType because it has a section name and isn't a string
```
Comment (dangerismycat) [inline docs/utils/mdHelper.js:48]: Can we fix these Flow issues?

Comment (rlingineni) [reply docs/utils/mdHelper.js:48]: This is one is weird because our Site Index is a mixed type array of `string` or `SiteIndex`. Normally in typescript, I'd cast it. I didn't see a good way to do that in Flow, hence the $FixMe

Let me know if there's a better way to cast this correctly?

File: docs/utils/mdHelper.js (hunk 3 above)
Comment (rlingineni) [reply docs/utils/mdHelper.js:43]: Good catch. That was me doing trying to solve some pass by reference issues. We don't need it anymore. Removed.

File: docs/markdown/get_started/developers/development_process.md [hunk 5]
```diff
@@ -1,21 +1,20 @@
...
-- **Install node**
-- **Install yarn**
-- **[Install Docker desktop](333)** \
-  You can also run the following command `brew install docker`
+- Install a code editor - we recommended [VS Code](https://code.visualstudio.com/download)
+- [Install nvm](https://github.com/creationix/nvm#install-script)
+- [Install node](https://github.com/nvm-sh/nvm#usage)
+- [Install yarn](https://classic.yarnpkg.com/en/docs/install)
+- [Install Docker desktop](https://docs.docker.com/get-docker/)
+  <Hint>You can also run the following command `brew install --cask docker`</Hint>
```
[the remark this answers is not in the document]
Comment (rlingineni) [reply docs/markdown/get_started/developers/development_process.md:13]: Tweaked the wording!

<Hint>To install Docker, try running `brew install --cask docker`</Hint>

File: docs/docs-components/MarkdownPage.js [hunk 6]
```diff
@@ -143,6 +160,17 @@ export default function MarkdownPage({ children, meta, pageSourceUrl }: Props):
     TwoCol: (props) => (
       <MainSection.Subsection columns={2}>{props.children}</MainSection.Subsection>
     ),
+    ThreeCol: (props) => (
```
[the remark this answers is not in the document]
Comment (rlingineni) [reply docs/docs-components/MarkdownPage.js:163]: added!

File: docs/docs-components/MarkdownPage.js [hunk 7]
```diff
@@ -110,6 +112,21 @@ export default function MarkdownPage({ children, meta, pageSourceUrl }: Props):
         <MainSection.Subsection title={props.children} marginBottom="compact" />
       </Box>
     ),
+    IllustrationCard,
+    ImgContainer: (props: {| src: string |}) => (
```
[the remark this answers is not in the document]
Comment (rlingineni) [reply docs/docs-components/MarkdownPage.js:116]: fixed in #2227

File: docs/markdown/get_started/developers/development_process.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (rlingineni) [reply docs/markdown/get_started/developers/development_process.md:13]: fixed in #2227

File: docs/utils/mdHelper.js (hunk 3 above)
Comment (rlingineni) [reply docs/utils/mdHelper.js:43]: fixed in #2227

File: docs/markdown/ios/avatar.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+<TwoCol>
+  <Group>
+    <ImgContainer src="https://i.pinimg.com/originals/c9/f0/c5/c9f0c597503b7d3ea1dc840ce1e3add8.png" alt="do use right image size"/>
+    <Do title="Do" />
+    Use round Avatars in the appropriate size for your need.
+  </Group>
+  <Group>
+    <ImgContainer src="https://i.pinimg.com/originals/8b/ea/f5/8beaf574ae9b9207700c63cbb6f33f33.png" alt="do not use not round images"/>
+    <Dont title="Don't" />
+    Use round Avatars in the appropriate size for your need.
```
Comment (ayeshakmaz) [inline docs/markdown/ios/avatar.md:38]: This seems wrong?

File: docs/markdown/ios/avatar.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+  </Group>
+  <Group>
+    <ImgContainer src="https://i.pinimg.com/originals/c9/f0/c5/c9f0c597503b7d3ea1dc840ce1e3add8.png" alt="do use avatar with names"/>
```
Comment (ayeshakmaz) [inline docs/markdown/ios/avatar.md:41]: This image is identical, is that expected?

File: docs/markdown/ios/avatar.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+People use Android’s accessibility features, such as TalkBack and dynamic text sizing to personalize how they interact with their device. Supporting these personalizations ensures that everyone has a great user experience. See Material Design and development documentation about accessibility for Android:
+
+[Accessible design on Android](https://material.io/design/usability/accessibility.html#understanding-accessibility)
+[Accessible development on Android](https://developer.android.com/guide/topics/ui/accessibility)
+
+## Variants
+
+### Size
+
+1. **xxl** **120px**
```
Comment (ayeshakmaz) [inline docs/markdown/ios/avatar.md:58]: I think it'd help legibility if we put the px equivalent inside parentheses

**Review by ayeshakmaz — CHANGES_REQUESTED** [review]

File: .prettierrc.json [hunk 11]
```diff
@@ -17,12 +17,6 @@
         "parser": "css",
         "singleQuote": false
       }
-    },
-    {
-      "files": "*.md",
```
Comment (dangerismycat) [inline .prettierrc.json:22]: We're no longer running Prettier on our .md files? Can we tweak the config instead?

File: docs/docs-components/IllustrationCard.js [hunk 12]
```diff
@@ -1,10 +1,11 @@
 // @flow strict
 import { Badge, Box, Card, Flex, Heading, TapArea, Text } from 'gestalt';
 import { type Node } from 'react';
+import illustrations from '../graphics/index.js';
 
-type Props = {|
+export type IllustrationCardProps = {|
   headingLevel: 2 | 3,
-  image: Node,
+  image: Node | string,
```
Comment (dangerismycat) [inline docs/docs-components/IllustrationCard.js:8]: Can we type this better than a string? It looks like it's a very short enum.

File: docs/docs-components/MarkdownPage.js [hunk 13]
```diff
@@ -142,6 +144,31 @@ const components = {
...
+  ImgContainer: ({ src, caption, alt }: {| src: string, caption?: string, alt?: string |}) => (
+    <Box>
+      <Box padding={8} rounding={2} borderStyle="sm" height="250px">
+        <Box position="relative" width="100%" height="100%">
+          <Image src={src} alt={alt} width="100%" height="100%" layout="fill" objectFit="contain" />
+        </Box>
+      </Box>
+      <Text size="300" align="start">
+        <Box as="figcaption" marginTop={3}>
+          {caption || ''}
```
Comment (dangerismycat) [inline docs/docs-components/MarkdownPage.js:156]: If `caption` is undefined, do we still want to render the Text and Box? Do we want a marginTop on no content?

Also: this looks like we instead want a Flex around the Image and Text with `gap={3}`. That gets the conditional margin for free.

File: docs/markdown/ios/avatar.md [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+## Usage guidelines
+
+<TwoCol>
+  <Group>
+    <Do title="Do" />
+      - To reflect a person, company, or brand within the product
+  </Group>
+  <Group>
+  <Dont title="Don't" />
+     - To represent a group of people, companies, and/or brands. use [AvatarGroup](https://gestalt.netlify.app/avatargroup) instead
```
Comment (dangerismycat) [inline docs/markdown/ios/avatar.md:18]: This is using the old domain. It should use the new one.

File: docs/markdown/ios/avatar.md [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+  <Group>
+    <ImgContainer src="https://i.pinimg.com/originals/c9/f0/c5/c9f0c597503b7d3ea1dc840ce1e3add8.png" alt="do use avatar with names"/>
+    <Do title="Do" />
+    Use the collaborators name nearby or in an alternative view if possible
+  </Group>
+</TwoCol>
+
+## Accessibility
+
+People use Android’s accessibility features, such as TalkBack and dynamic text sizing to personalize how they interact with their device. Supporting these personalizations ensures that everyone has a great user experience. See Material Design and development documentation about accessibility for Android:
```
Comment (dangerismycat) [inline docs/markdown/ios/avatar.md]: This is for iOS, yes? Seems odd to have Android content here

File: docs/markdown/ios/avatar.md (hunk 10 above)
Comment (dangerismycat) [inline docs/markdown/ios/avatar.md]: Can we present this as a table instead?

File: docs/markdown/ios/avatar.md [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+3. **lg** **44px**
+4. **md** **32px**
+5. **sm** **24px**
+6. **xs** **16px**
+
+<ImgContainer src="https://i.pinimg.com/originals/30/7a/de/307adebcacf4fca359de788e6a077329.png" alt="different avatar sizes"/>
+
+### Without an image
+
+If there is no image source provided for the Avatar, the first character of the name provided will be used as a placeholder.
```
Comment (dangerismycat) [inline docs/markdown/ios/avatar.md:69]: We don't use articles with component names — so instead of "…for the Avatar", just "…for Avatar".

File: docs/markdown/ios/avatar.md [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+---
+
+<ImgContainer src="https://i.pinimg.com/originals/30/7a/de/307adebcacf4fca359de788e6a077329.png" alt="avatar hero image"/>
+
+## Usage guidelines
+
+<TwoCol>
+  <Group>
+    <Do title="Do" />
+      - To reflect a person, company, or brand within the product
```
Comment (dangerismycat) [inline docs/markdown/ios/avatar.md:14]: super nit: as much as it physically pains me, we don't use Oxford commas in our style guide for the docs (and all of Pinterest) 😭

File: docs/markdown/ios/overview.md [hunk 18]
```diff
@@ -0,0 +1 @@
+Coming Soon!!!
```
Comment (dangerismycat) [inline docs/markdown/ios/overview.md:1]: Is this change necessary? We're losing the page title (and I suspect the `h1` necessary for accessibility purposes)
Screen Shot 2022-08-22 at 11 08 50 AM
Screen Shot 2022-08-22 at 11 08 39 AM

File: docs/graphics/index.js [hunk 19]
```diff
@@ -0,0 +1,10 @@
+// @flow strict
+import AvatarGroup from './general/AvatarGroup.svg';
+
+export type IllustrationTypes = 'avatar-group';
```
Comment (dangerismycat) [inline docs/graphics/index.js:4]: Rather than repeating the keys, check out [this utility type](https://flow.org/en/docs/types/utilities/#toc-keys)

File: docs/markdown/get_started/developers/development_process.md [hunk 20]
```diff
@@ -173,6 +173,7 @@ A Gestalt Core maintainer will add a semver label (patch release / minor release
 
 When pushing new changes to GitHub, your PR title should be aligned with the scope of your work. If your goal was to change the default color of a component, keep the scope of changes to that specific task and word the title to exactly reflect those changes.
 
+
```
Comment (dangerismycat) [inline docs/markdown/get_started/developers/development_process.md:176]: nit: you may want to revert this change for a cleaner git history

**Review by dangerismycat — APPROVED** [review]

**Review by ayeshakmaz — APPROVED** [review]
